src/System.Net.Primitives/src/System/Net/NetworkCredential.cs

```diff
@@ -76,14 +85,15 @@ public string UserName
         /// </devdoc>
         public string Password
         {
-            get
-            {
-                return InternalGetPassword();
-            }
-            set
-            {
-                _password = value;
-            }
+            get { return _password; }
+            set { _password = value ?? string.Empty; }
+        }
+
+        [CLSCompliant(false)]
+        public SecureString SecurePassword
+        {
+            get { return _securePassword.Copy(); } 
+            set { _securePassword = value != null ? value.Copy() : new SecureString(); }
```

Comment:
We should be disposing of the old password, e.g.
```C#
set
{
    SecureString old = _securePassword;
    _securePassword = value != null ? value.Copy() : new SecureString();
    old.Dispose();
}
```
If we don't, it'll eventually get cleaned up due to finalization, but it's better to be aggressive about it.